docs/best_practice/systables.txt

```diff
@@ -0,0 +1,161 @@
+the current master along with several other settings. To list all
+available data, run::
+
+    cr> SHOW COLUMNS IN cluster FROM sys;
+    ...
+
+While **sys.cluster** contains information about the cluster as a whole,
+**sys.nodes** maintains more detailed information about each Crate
+instance. This can be useful to track down misbehaving data nodes since
+its CPU is overloaded or it has an outdated Java version::
+
+    cr> SELECT name, load['1'], os_info['jvm']['version'] FROM sys.nodes;
+    ...
+
+To list all nodes using more than 90 per cent of the memory, type::
+
+    cr> SELECT * FROM sys.nodes WHERE mem['used_percent'] > 90;
+    ...
+
+The table also contains the performance metrics like the load average,
+disk, memory, heap, or network throughput. Running::
+
+    cr> SHOW columns IN nodes FROM sys;
+    ...
+
+lists all available attributes. This object has the same structure as
+the **_node** attribute of **sys.operations** from the previous
+section.
+
+Step 4: Insights about Partitions, Shards, and Replication
+==========================================================
+
+Crate divides the rows of each table into shards that are distinctively
+distributed to all nodes in your cluster. Replication uses the same
+mechanism to add redundancy and thus resilience to your data. While most
+of the time Crate takes transparently care of distributing and
+replicating the shards, it’s useful during troubleshooting to actually
+find out some more about these data structures. The **sys.shards** table
+provides access to the status and size of shards, its names and IDs::
+
+    cr> SHOW COLUMNS IN shards FROM sys;
+    ...
+
+The cluster state is somewhat delicate when nodes join or leave, since
+in those situations shards have to be rearranged to ensure that each of
+them is replicated to different nodes. As long as the **state**
+attribute is ‘STARTED’ for all shards, the cluster is in a stable state,
+otherwise Crate is occupied with some background activity. The cluster
+state indicators on the admin UI evaluate these values as well.
+
+The **sys.shards** table contains even more information about the
+rebalancing activities. Whenever Crate needs to copy one shard to
+another node -- since that may be necessary to ensure there are enough
+replica of it distributed in the cluster -- you can monitor the progress
+of that operation with the **recovery** object::
+
+    cr> SELECT DISTINCT partition_ident, recovery['size']['percent'], recovery['total_time'] FROM sys.shards WHERE primary;
+    ...
+
+To find out on which specific node a shard is stored, also use the
+_node object that is available for this table. As an example::
+
+    cr> SELECT _node['hostname'], table_name, num_docs FROM sys.shards ORDER BY num_docs DESC LIMIT 3;
+    ...
+
+list the hosts and tables with the highest number of rows inside a
+single shard. There’s a lot of additional information available for
+shards. To find out more about that, see our documentation.
```

Comment:
a detail link here would be nice, since the user is already in the documentation